spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteDataFilesSparkAction.java

```diff
@@ -337,14 +337,21 @@ private Result doExecuteWithPartialProgress(
         commitManager.service(groupsPerCommit);
     commitService.start();
 
+    List<FileGroupFailureResult> rewriteFailures = Lists.newCopyOnWriteArrayList();
```

Comment:
In general this would probably be an expensive data-structure for this operation since we never traverse. I don't think this is ever going to be that huge, but let's use something a bit more efficient. 

I think something like 
ConcurrentLinkedQueue would be fine here, and then we could just create a list out of it when we are done

Reply:
Just to note, I don't think we actually would have performance issues with copy on write array list, it just seems a bit wasteful to me in this context.

Reply:
After reading the introduction of newCopyOnWriteArrayList and ConcurrentLinkedQueue, it is true that the latter is more efficient, thank you.